packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/preview/index.ts

```diff
@@ -29,26 +29,49 @@ export default class CardPayDepositWorkflowPreviewComponent extends Component<Ca
   @reads('args.workflowSession.state.spendFaceValue')
   declare faceValue: number;
 
-  @tracked errorMessage = '';
+  @tracked error = '';
 
   @task *issueTask(): TaskGenerator<void> {
+    this.error = '';
+
     let { workflowSession } = this.args;
-    yield this.hubAuthentication.ensureAuthenticated();
-    // yield statements require manual typing
-    // https://github.com/machty/ember-concurrency/pull/357#discussion_r434850096
-    let customization: Resolved<PrepaidCardCustomization> = yield taskFor(
-      this.cardCustomization.createCustomizationTask
-    ).perform({
-      issuerName: workflowSession.state.issuerName,
-      colorSchemeId: workflowSession.state.colorScheme.id,
-      patternId: workflowSession.state.pattern.id,
-    });
-    yield taskFor(this.layer2Network.issuePrepaidCard)
-      .perform(this.faceValue, customization.did)
-      .then((address: string) => {
-        this.args.workflowSession.update('prepaidCardAddress', address);
-        this.args.onComplete();
+    try {
+      yield this.hubAuthentication.ensureAuthenticated();
+
+      // yield statements require manual typing
+      // https://github.com/machty/ember-concurrency/pull/357#discussion_r434850096
+      let customization: Resolved<PrepaidCardCustomization> = yield taskFor(
+        this.cardCustomization.createCustomizationTask
+      ).perform({
+        issuerName: workflowSession.state.issuerName,
+        colorSchemeId: workflowSession.state.colorScheme.id,
+        patternId: workflowSession.state.pattern.id,
       });
+
+      yield taskFor(this.layer2Network.issuePrepaidCard)
+        .perform(this.faceValue, customization.did)
+        .then((address: string) => {
+          this.args.workflowSession.update('prepaidCardAddress', address);
+          this.args.onComplete();
+        });
+    } catch (e) {
+      let insufficientFunds = e.message.startsWith(
+        'Safe not have enough balance to make prepaid card(s).'
```

Comment:
my bad, can we fix this error message so that the grammar is correct? "Safe _does_ not have enough balance to make prepaid card(s)"

Reply:
Will do!